packages/flutter/lib/src/gestures/tap.dart

```diff
@@ -18,12 +18,15 @@ import 'recognizer.dart';
 class TapDownDetails {
   /// Creates details for a [GestureTapDownCallback].
   ///
-  /// The [globalPosition] argument must not be null.
-  TapDownDetails({ this.globalPosition = Offset.zero })
-    : assert(globalPosition != null);
+  /// Arguments [globalPosition] and [buttons] must not be null.
+  TapDownDetails({ this.globalPosition = Offset.zero, this.buttons = 0 })
```

Comment:
I think before we make this change we should update the Android and iOS logic to treat fingers-touching-surface and stylus-touching-surface as button=1.